sensors/test_deps.go

```diff
@@ -356,6 +381,48 @@ func getOdometerWithErroringFunctions() *inject.MovementSensor {
 	return odometer
 }
 
+func getReplayOdometer(testTime string) *inject.MovementSensor {
+	odometer := &inject.MovementSensor{}
+	odometer.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+		md := ctx.Value(contextutils.MetadataContextKey)
+		if mdMap, ok := md.(map[string][]string); ok {
+			mdMap[contextutils.TimeRequestedMetadataKey] = []string{testTime}
+		}
+		return TestPosition, 1.2, nil
+	}
+	odometer.OrientationFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+		md := ctx.Value(contextutils.MetadataContextKey)
+		if mdMap, ok := md.(map[string][]string); ok {
+			mdMap[contextutils.TimeRequestedMetadataKey] = []string{testTime}
+		}
+		return TestOrientation, nil
+	}
+	odometer.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{
+			PositionSupported:    true,
+			OrientationSupported: true,
+		}, nil
+	}
+	return odometer
+}
+
+func getFinishedReplayOdometer() *inject.MovementSensor {
+	odometer := &inject.MovementSensor{}
+	odometer.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+		return geo.NewPoint(0, 0), 0.0, replay.ErrEndOfDataset
```

Comment:
I prefer `math.Nan` for things that actually mean something physical.

Reply:
So you would want `geo.NewPoint(math.Nan, math.Nan)`? I suggested nil because the return of Position is a `geo.Point` pointer

Reply:
Note: I also just checked RDK and movement sensors that dont implement Position (such as imuwit) returns geo.NewPoint(0,0). This isn't my favorite pattern to support as if an error is returned I'd like the result to be inaccessible and not take up any memory (so nil)

Reply:
^ No, not all; our preferred pattern with an error is `math.Nan`. We just haven't gotten around to changing them yet. So you're looking at a bad example. Also the ones that truly don't have out ErrUnimplemented returned, so you can't actually get info out of that client call at all.

The one that does implement Position and has an error associated with it is GPS, where if it is erroring, we return math.NaN. 

Also the problem here is if you return it as nil, and someone else uses the geoPoint library that we're wrapping on that nil pointer - like `position.MidPointTo` you get panics all over the place. I'd rather give NaNs.

Reply:
The different here is - the witimu can never return position - it doesn't have that data or a chance at returning that data at all, so we use the movement sensor error unimplemented and a PositionReporting: false Property to make sure that doesn't show up in the sdks or the rc card. So you just get an error unimplemented if you try to get any of that information for business level logic out of the wit imu.

Reply:
^ Since those aren't returned pointers like the *geo.Point return from Position: I'm not as concerned about the r3 library accessing them after they're returned. So it is up to you in this pr.

 Is cartographer seeing zero values and taking them as meaningful? That's a question for another day, maybe it has a safeguard against a constant stream of zero data, some algorithms do.

Reply:
TBR (To be researched) about carto's handling of constant streams of zeros